Adds new Partner Highlight blog on semantic search. (#2884)
opensearch-project/project-website: _posts/2024-05-30-do-you-need-semantic-search.markdown

---
layout: post
title:  "Do you need semantic search?"
authors:
  - dtaivpp
date:   2024-05-30 11:45 -0700
categories:
  - technical-posts
meta_keywords: opensearch hybrid search, opensearch semantic search, vector search
meta_description: You might be asking yourself do you really need semantic search? I've been asked this question dozens of times so lets dive in and see what semantic search is and why you might want to use it. 
SemanticSearch: lpQiJGpeeWU
---

> I get ask this question all the time. Do I need semantic search? It's a good question but it's not the right question to start with. We need to first understand what semantic search does for us. Then we can dive into whether or not it's a good fit for your use case.

Reviewer: ```suggestion
I get asked this question all the time: Do I need semantic search? It's a good question, but it's not the right question to start with. You need to first understand what semantic search can do for you. Then you can dive into whether or not it's a good fit for your use case.
```

> I've answered both questions in my [most recent blog](https://tippybits.com/should-you-be-doing-vector-search/) and its video counterpart linked below. These answer the following questions: 

Reviewer: ```suggestion
I've answered both questions in my [most recent blog post](https://tippybits.com/should-you-be-doing-vector-search/) and its video counterpart, linked below. These answer the following questions: 
```

1. How does OpenSearch find documents out of the box?

> 2. Why these traditional search methods don't always work. 

Reviewer: ```suggestion
2. Why don't these traditional search methods always work? 
```

> 3. When and how to use semantic search in OpenSearch. 

Reviewer: ```suggestion
3. When and how should semantic search be used in OpenSearch? 
```

> To help I created some additional resources to help you get started! [This GitHub Gist](https://gist.github.com/dtaivpp/c587d99a2cab441eba0314534ae87c86) will help you create an OpenSearch docker compose environment. Also, I have created [this Gist](https://gist.github.com/dtaivpp/d7e8d8a3ee5debaf896ed2f45b915ad3) with all the code needed to create your own semantic search pipeline right from OpenSearch Dashboards. I hope you enjoy them and if your company needs help getting started with semantic search they can reach out to me [on my website](https://tippybits.com/services)!

Reviewer: ```suggestion
I have created some additional resources to help you get started! [This GitHub Gist](https://gist.github.com/dtaivpp/c587d99a2cab441eba0314534ae87c86) will help you create an OpenSearch Docker Compose environment. I have also created [this Gist](https://gist.github.com/dtaivpp/d7e8d8a3ee5debaf896ed2f45b915ad3) with all the code needed to create your own semantic search pipeline right from OpenSearch Dashboards. I hope you enjoy them, and if your organization needs help getting started with semantic search, you can reach out to me [on my website](https://tippybits.com/services).
```

> To help I created some additional resources to help you get started! [This GitHub Gist](https://gist.github.com/dtaivpp/c587d99a2cab441eba0314534ae87c86) will help you create an OpenSearch docker compose environment. Also, I have created [this Gist](https://gist.github.com/dtaivpp/d7e8d8a3ee5debaf896ed2f45b915ad3) with all the code needed to create your own semantic search pipeline right from OpenSearch Dashboards. I hope you enjoy them and if your company needs help getting started with semantic search they can reach out to me [on my website](https://tippybits.com/services)!

Reviewer: Minor suggestion: "find them useful" instead of "enjoy them"?
